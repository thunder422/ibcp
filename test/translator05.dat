####################################################
#  Translator Test #5: Command Tests (2010-05-29)  #
####################################################

#  LET tests
LET A=0.0
LET A%=0
LET A$=""
LET A=B=C=0.0
LET A,B,C=0.0
LET A%=B%=C%=0
LET A%,B%,C%=0
LET A$=B$=C$=""
LET A$,B$,C$=""
LET LEFT$(A$,1) = B$
LET LEFT$(A$,1) = B$ + C$
LET LEFT$(A$,1),RIGHT$(B$,1) = C$
LET A$,LEFT$(B$,1) = C$

# LET error tests
LET LET A=0
A = 0 LET 5

PRINT
