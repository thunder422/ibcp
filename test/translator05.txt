
Input: LET A=0.0
Output: A<ref> 0.0 =Assign'LET' 

Input: LET A%=0
Output: A%<ref> 0% =Assign%'LET' 

Input: LET A$=""
Output: A$<ref> "" =Assign$'LET' 

Input: LET A=B=C=0.0
Output: A<ref> B C = 0.0% =% CvtDbl =Assign'LET' 

Input: LET A,B,C=0.0
Output: A<ref> B<ref> C<ref> 0.0 =AssignList'LET' 

Input: LET A%=B%=C%=0
Output: A%<ref> B% C% =% 0% =% =Assign%'LET' 

Input: LET A%,B%,C%=0
Output: A%<ref> B%<ref> C%<ref> 0% =AssignList%'LET' 

Input: LET A$=B$=C$=""
                    ^^-- expected numeric expression

Input: LET A$,B$,C$=""
Output: A$<ref> B$<ref> C$<ref> "" =AssignList$'LET' 

Input: LET LEFT$(A$,1) = B$
Output: A$<ref> 1% B$ AssignLeft'LET' 

Input: LET LEFT$(A$,1) = B$ + C$
Output: A$<ref> 1% B$ C$ +$ AssignLeft'LET' 

Input: LET LEFT$(A$,1),RIGHT$(B$,1) = C$
Output: A$<ref> 1% B$<ref> 1% C$ AssignKeepRight AssignLeft'LET' 

Input: LET A$,LEFT$(B$,1) = C$
Output: A$<ref> B$<ref> 1% C$ AssignKeepLeft =Assign$'LET' 

Input: LET LET A=0
           ^^^-- expected item for assignment

Input: A = 0 LET 5
             ^^^-- expected operator or end-of-statement

Input: PRINT
Output: PRINT 

