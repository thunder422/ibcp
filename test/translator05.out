
Input: LET A=0.0
Output: LET A = 0.0 

Input: LET A%=0
Output: LET A% = 0 

Input: LET A$=""
Output: LET A$ = "" 

Input: LET A=B=C=0.0
Output: LET A = B = C = 0.0 

Input: LET A,B,C=0.0
Output: LET A, B, C = 0.0 

Input: LET A%=B%=C%=0
Output: LET A% = B% = C% = 0 

Input: LET A%,B%,C%=0
Output: LET A%, B%, C% = 0 

Input: LET A$=B$=C$=""
                    ^^-- expected numeric expression

Input: LET A$,B$,C$=""
Output: LET A$, B$, C$ = "" 

Input: LET LEFT$(A$,1) = B$
Output: LET LEFT$(A$, 1) = B$ 

Input: LET LEFT$(A$,1) = B$ + C$
Output: LET LEFT$(A$, 1) = B$ + C$ 

Input: LET LEFT$(A$,1),RIGHT$(B$,1) = C$
Output: LET LEFT$(A$, 1), RIGHT$(B$, 1) = C$ 

Input: LET A$,LEFT$(B$,1) = C$
Output: LET A$, LEFT$(B$, 1) = C$ 

Input: LET LET A=0
           ^^^-- expected item for assignment

Input: A = 0 LET 5
             ^^^-- expected operator or end-of-statement

Input: PRINT
Output: PRINT 

